Lib/fontTools/ttLib/ttFont.py

```diff
@@ -4,12 +4,7 @@
 from fontTools.misc.textTools import Tag, byteord, tostr
 from fontTools.misc.loggingTools import deprecateArgument
 from fontTools.ttLib import TTLibError
-from fontTools.ttLib.ttGlyphSet import (
-    _TTGlyphSet, _TTGlyph,
-    _TTGlyphCFF, _TTGlyphGlyf,
```

Comment:
I wonder if we should keep the `_TTGlyphSet`, `_TTGlyphCFF` and `_TTGlyphGlyf` symbols here, even if unused. There may be client code that may expect to find these in `ttFont` module, as they were there before things where moved to the new `ttGlyphSet`. I know they are underscore prefixed and thus private, but one may still be using them in typing annotations for function parameters/return values that include those.

Reply:
The only reason `_TTGlyph` is there is because it's used as a typing thing in the remove overlaps module. Even there I wonder: does it need to be this concrete (but abstract) class, or should we define our own abstract typing declarations? "What is a glyph set" / "What is a glyph".

I don't want to keep more "just in case", unless there's actual client code that uses these (but really shouldn't).

Reply:
> should we define our own abstract typing declarations? "What is a glyph set" / "What is a glyph".

yeah maybe we could do that, perhaps using typing.Protocol.

Reply:
e.g. I found code like this one in our internal codebase:

```python
  def to_ttglyph(self, character: Text) -> ttFont._TTGlyph:
    """Converts a character to its associated ttFont._TTGlyph.
    ...
```

what should client code do in this case?

Reply:
I don't really know. We could rename it `TTGlyph`, and the client code should ideally import it from ttGlyphSet.py instead of from ttFont.py.

What aspects of the `_TTGlyph` instance does the caller of `to_ttglyph()` actually use? Is it expecting all kinds of internals of that object, or does it expect draw/drawPoints methods and maybe the metrics attrs?